best-practices-zarr-ndarray.md

```diff
@@ -0,0 +1,536 @@
+- [Raster Extension v2.x](https://github.com/stac-extensions/raster)
+- [Electro Optical (EO) Extension v2.x](https://github.com/stac-extensions/eo)
+- [Projection Extension v2.x](https://github.com/stac-extensions/projection)
+
+## Glossary
+
+### Zarr Terminology
+
+See the [Zarr specification](https://zarr-specs.readthedocs.io/en/latest/v3/core/#concepts-and-terminology) for complete terminology.
+
+Key terms:
+
+- **Zarr Store**: A storage system that can hold a Zarr hierarchy (e.g., a directory, zip file, or object store prefix)
+- **Group**: A node in a Zarr hierarchy that can contain arrays and/or other groups
+- **Array**: A multidimensional array with chunked storage
+- **Chunk**: A contiguous region of an array that is stored as a single object
+- **Consolidated Metadata**: A single metadata document containing the metadata for all arrays and groups in a store
+
+### STAC-Specific Concepts
+
+- **Variable**: A named data array within a Zarr group, typically representing a measured or derived quantity
+- **Dimension**: An axis of a multidimensional array (e.g., x, y, time, band)
+- **Coordinate Variable**: An array that provides coordinate values along a dimension
+- **Data Variable**: An array containing measured or derived data values
+
+## General Rules
+
+### Asset Organization
+
+1. **A Zarr asset SHALL be a group containing one or more arrays**
+   
+   This is equivalent to an xarray Dataset, ensuring that Zarr assets are accessible data containers rather than empty hierarchies.
+
+2. **The Zarr store SHOULD be referenced with a link using the `"store"` relationship**
+   
+   ```json
+   "links": [
+     {
+       "rel": "store",
+       "href": "s3://bucket/path/data.zarr",
+       "type": "application/vnd+zarr; version=2",
+       "title": "Zarr Store"
+     }
+   ]
+   ```
+
+3. **Assets can reference Zarr groups or individual arrays within the store**
+   
+   Assets may point to:
+   - The root store
+   - Intermediate groups
+   - Individual data variables/arrays
+   
+   The appropriate level depends on how users will access the data.
+
+4. **Asset roles SHOULD clearly indicate the type of reference**
+   
+   - `"store"`: Reference to a complete Zarr store
+   - `"group"`: Reference to a group within the store
+   - `"data"`: Reference to individual data arrays
+
+### Link Relationships
+
+**Store Link Relationship**
+
+A new link relationship `"store"` is used to reference Zarr stores from Collections or Items:
+
+```json
+"links": [
+  {
+    "rel": "store",
+    "href": "s3://bucket/path/data.zarr",
+    "type": "application/vnd+zarr; version=2",
+    "title": "Zarr Store"
+  }
+]
+```
+
+**TODO**: Propose this relationship for inclusion in the storage STAC extension.
+
+**Media Type**
+
+The media type for Zarr should include version information as a parameter:
+
+- Zarr v2: `"application/vnd+zarr; version=2"`
+- Zarr v3: `"application/vnd+zarr; version=3"`
```

Comment:
Should probably also add those in the table in [asset and link best practices](https://github.com/radiantearth/stac-best-practices/blob/main/best-practices-asset-and-link.md#working-with-media-types)